### DISCUSSION TITLE
httpd as non-admin on macOS

### DISCUSSION THREAD

**Discussion by gitdev-f11** [Homebrew/discussions#4209] [Everyday usage]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 3.6.21
ORIGIN: https://github.com/Homebrew/brew
HEAD: 932d2cf3b77c9439a57b6a43577fc8d3b6399a62
Last commit: 7 days ago
Core tap origin: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 13913785304b17c76624838dce990305941d0631
Core tap last commit: 13 hours ago
Core tap branch: master
Core tap JSON: 13 Feb 12:23 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 10
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: 10-core 64-bit arm_blizzard_avalanche
Clang: 14.0.0 build 1400
Git: 2.37.1 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 7.86.0 => /usr/bin/curl
macOS: 13.2.1-arm64
CLT: 14.2.0.0.1.1668646533
Xcode: 14.2
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

Hi.
I have installed httpd + PHP via homebrew on a new M2 Macbook. As a user with admin rights, which all works.
However, I am usually logged in as a **user without admin rights** and want to access the web server. But that doesn't work, the browser always replies 'no connection to server'.
I have already tried all the variations of localhost, port, etc.

Is there an option to connect the web server with a non-admin account?

btw: what do I have to do to access the rest of brew terminal functions as a non-admin ? su admin ?

thanks for answers

**Comment by SMillerDev** [comment]
How do you start httpd? What port do you have it configured with? Is your admin user still logged in?

> what do I have to do to access the rest of Brei's terminal functions as a non-admin ? su admin ?

Yes
> If you need to run Homebrew in a multi-user environment, consider creating a separate user account specifically for use of Homebrew.

https://docs.brew.sh/FAQ#why-does-homebrew-say-sudo-is-bad

**Reply by gitdev-f11** [reply]
> How do you start httpd?

brew services start/stop/restart httpd

> What port do you have it configured with?

80 and 8080

> Is your admin user still logged in?

no, admin is logged out !

> [...]

**Reply by SMillerDev** [reply]
> brew services start/stop/restart httpd

That starts the service for the user.


> no, admin is logged out !

So the service isn't running because of that.

**Reply by gitdev-f11** [reply]
ok, what would be a workaround to keep the process running ?
or can I also start it as non-admin ?

**Reply by gitdev-f11** [reply]
would it be a solution to install brew in the user directory ? or better assign a usergroup to brew and the non-admin user to it ?

**Reply by SMillerDev** [reply]
https://docs.brew.sh/Manpage#services-subcommand

**Reply by gitdev-f11** [reply]
good old rtfm ;)
it works' !

THX
